File: Lessons/04_CountingElements_MissingInteger.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
    # write your code in Ruby 2.2
    hash = {}
    a.each do |aa|
      hash[aa] = true if aa > 0
    end 
    1.upto(10**10) do |i|
      return i if hash.has_key?(i) == false 
    end 
end

File: Python/04_CountingElements_MissingInteger.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    dic = {}
    for a in A:
        if a > 0: dic[a] = True  
    
    for i in range(1, 10 ** 10):
        if i not in dic: return i 
# end of func
